### PR TITLE
Add common policheck steps

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Adding common policheck steps to enable policheck run on PR validations.

### CODE REVIEW

File: eng/common/pipelines/templates/steps/policheck.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+
+parameters:
+  ExclusionDataBaseFileName: ''
+  TargetDirectory: ''
+  PublishAnalysisLogs: false
+  PoliCheckBlobSAS: '$(azuresdk-policheck-blob-SAS)'
```
Comment (benbp) [inline eng/common/pipelines/templates/steps/policheck.yml:6]: Do we need to add variable group 95 to the pipeline generators for this?

File: eng/common/pipelines/templates/steps/policheck.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/policheck.yml:1]: nit empty line

File: eng/common/pipelines/templates/steps/policheck.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    displayName: 'Run PoliCheck'
+    inputs:
+      targetType: F
+      targetArgument: '$(Build.SourcesDirectory)/${{ parameters.TargetDirectory }}'
+      result: PoliCheck.sarif
+      optionsFC: 0
+      optionsXS: 1
+      optionsPE: 1|2|3|4
+      optionsRulesDBPath: "$(Build.BinariesDirectory)/${{ parameters.ExclusionDataBaseFileName }}.mdb"
+      optionsUEPATH: "$(Build.SourcesDirectory)/eng/guardian-tools/policheck/PolicheckExclusions.xml"
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/policheck.yml:24]: This should probably be a parameter as well with a default set to this value.

File: eng/common/pipelines/templates/steps/policheck.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  - pwsh: |
+      azcopy copy "https://azuresdkartifacts.blob.core.windows.net/policheck/${{ parameters.ExclusionDataBaseFileName }}.mdb?${{ parameters.PoliCheckBlobSAS }}" `
+      "$(Build.BinariesDirectory)"
+    displayName: 'Download PoliCheck Exclusion Database'
+
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-policheck.PoliCheck@2
+    displayName: 'Run PoliCheck'
+    inputs:
+      targetType: F
+      targetArgument: '$(Build.SourcesDirectory)/${{ parameters.TargetDirectory }}'
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/policheck.yml:18]: When we start going to enable this in PR validations will this correctly work?

File: eng/common/pipelines/templates/steps/policheck.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@2
+    displayName: 'Post Analysis (PoliCheck)'
+    inputs:
+      GdnBreakAllTools: false
+      GdnBreakGdnToolPoliCheck: true
+      GdnBreakGdnToolPoliCheckSeverity: Warning
+    continueOnError: true
+
+  - ${{ if eq(parameters.PublishAnalysisLogs, 'true') }}:
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/policheck.yml:34]: What cases do you expect we will not want to publish the logs?

Comment (chidozieononiwu) [reply eng/common/pipelines/templates/steps/policheck.yml:34]: When using this in the aggreagte reports pipeline we might want to just pulish the log in one step together with credscan and any of the other secure development tools.

File: eng/common/pipelines/templates/steps/policheck.yml (hunk 4 above)
Comment (chidozieononiwu) [reply eng/common/pipelines/templates/steps/policheck.yml:18]: Yes this is a way to scope the run to the package triggering the PR. It should not be a problem.

File: eng/common/pipelines/templates/steps/policheck.yml (hunk 5 above)
Comment (weshaggard) [reply eng/common/pipelines/templates/steps/policheck.yml:34]: ok

File: eng/common/pipelines/templates/steps/policheck.yml (hunk 1 above)
Comment (chidozieononiwu) [reply eng/common/pipelines/templates/steps/policheck.yml:6]: How did you determine the number 95?

File: eng/common/pipelines/templates/steps/policheck.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
+parameters:
+  ExclusionDataBaseFileName: ''
+  TargetDirectory: ''
+  PublishAnalysisLogs: false
+  PoliCheckBlobSAS: "$(azuresdk-policheck-blob-SAS)"
+  UePath: "$(Build.SourcesDirectory)/eng/guardian-tools/policheck/PolicheckExclusions.xml"
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/policheck.yml:6]: What does `Ue` mean? Should we just call this `ExclusionFilePath`

**Review by weshaggard — APPROVED** [review]

File: eng/common/pipelines/templates/steps/policheck.yml (hunk 1 above)
Comment (weshaggard) [reply eng/common/pipelines/templates/steps/policheck.yml]: I was wondering that as well :) I assume he looked up the variable group that defines this.

Comment (chidozieononiwu) [reply eng/common/pipelines/templates/steps/policheck.yml]: I tried that, I still didn't see any number anywhere.

Comment (weshaggard) [reply eng/common/pipelines/templates/steps/policheck.yml]: The only way I have found to find the variable group id is to look at the URL when you are viewing a variable group.

Comment (chidozieononiwu) [reply eng/common/pipelines/templates/steps/policheck.yml]: Thanks @weshaggard . You can also find it using azure cli `az pipelines variable-group list --group-name "azuresdkartifacts SAS tokens`. But Url is definitly easier.

**Pull request merged**